File: vital-configs/src/main/java/me/vitalframework/configs/VitalConfigService.java
package me.vitalframework.configs;

import lombok.NonNull;
import me.vitalframework.Vital;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.List;

@Service
public class VitalConfigService {
    @NonNull
    public Collection<? extends VitalConfig> getConfigs(Class<? extends VitalConfig> vitalConfigClass) {
        try {
            return Vital.getContext().getBeansOfType(vitalConfigClass).values();
        } catch (Exception e) {
            return List.of();
        }
    }

    @NonNull
    public Collection<? extends VitalConfig> getConfigs() {
        return getConfigs(VitalConfig.class);
    }


    public <T extends VitalConfig> T getConfig(@NonNull Class<T> vitalConfigClass) {
        try {
            return Vital.getContext().getBean(vitalConfigClass);
        } catch (Exception e) {
            return null;
        }
    }
}

File: vital-configs/src/main/java/me/vitalframework/configs/VitalConfigService.kt
package me.vitalframework.configs

import me.vitalframework.Vital.context
import org.springframework.stereotype.Service

@Service
class VitalConfigService {
    fun getConfigs(vitalConfigClass: Class<VitalConfig>) =
        try {
            context.getBeansOfType(vitalConfigClass).values
        } catch (e: Exception) {
            mutableListOf()
        }

    fun getConfigs() = getConfigs(VitalConfig::class.java)

    fun <T : VitalConfig> getConfig(vitalConfigClass: Class<T>) =
        try {
            context.getBean<T?>(vitalConfigClass)
        } catch (e: Exception) {
            null
        }
}